pipelines/build/regeneration/build_pipeline_generator.groovy

```diff
@@ -1,59 +1,143 @@
+import java.nio.file.NoSuchFileException
+import groovy.json.JsonSlurper
+import groovy.json.JsonOutput
+//TODO: Change me
+String DEFAULTS_FILE_URL = (params.DEFAULTS_URL) ?: "https://raw.githubusercontent.com/M-Davies/openjdk-build/parameterised_everything/pipelines/defaults.json"
+
 node('master') {
-  def retiredVersions = [9, 10, 12, 13, 14]
+  timestamps {
+    // Retrieve Defaults
+    def get = new URL(DEFAULTS_FILE_URL).openConnection()
+    Map<String, ?> DEFAULTS_JSON = new JsonSlurper().parseText(get.getInputStream().getText()) as Map
+    if (!DEFAULTS_JSON) {
+      throw new Exception("[ERROR] No DEFAULTS_JSON found at ${DEFAULTS_FILE_URL}. Please ensure this path is correct and it leads to a JSON or Map object file.")
+    }
 
-  (8..30).each({javaVersion -> 
+    def retiredVersions = [9, 10, 12, 13, 14]
```

Comment:
should this be parameterised?

Reply:
It should be automatic, the user shouldn't have to enter it as a param at all. Which is why I'm working on https://github.com/AdoptOpenJDK/openjdk-api-v3/pull/314 🙂